lib/code_corps_web/channels/conversation_channel.ex

```diff
@@ -1,20 +1,21 @@
 defmodule CodeCorpsWeb.ConversationChannel do
   use Phoenix.Channel
 
-  alias CodeCorps.{Conversation, Repo}
+  alias CodeCorps.{Conversation, Policy, Repo, User}
+  alias Phoenix.Socket
 
-  def join("conversation:" <> id, _params, %{assigns: %{current_user: user}} = socket) do
-    conversation = Conversation |> Repo.get(id)
+  @spec join(String.t, map, Socket.t) :: {:ok, Socket.t} | {:error, map}
+  def join("conversation:" <> id, %{}, %Socket{} = socket) do
+    with %Conversation{} = conversation <- Conversation |> Repo.get(id),
+         %User{} = current_user <- socket.assigns[:current_user],
+         {:ok, :authorized} <- current_user |> Policy.authorize(:show, conversation, %{}) do
 
-    if user |> authorized?(conversation) do
-      {:ok, socket}
+        {:ok, socket}
     else
-      {:error, %{reason: "unauthorized"}}
+      nil -> {:error, %{reason: "unauthenticated"}}
+      {:error, :not_authorized} -> {:error, %{reason: "unauthorized"}}
     end
   end
-  def join(_, _, _) do
-    {:error, %{reason: "unauthenticated"}}
-  end
```

Comment:
If the `Repo.get` returns `nil` then we return `{:error, %{reason: "unauthenticated"}}`. I think this changes the behavior we had wanted originally, which is determining whether the user is authenticated or not, hence the pattern match on `assigns`.